app/models/manageiq/providers/hawkular/common/event_catcher/runner_mixin.rb

```diff
@@ -0,0 +1,50 @@
+module ManageIQ::Providers::Hawkular::Common::EventCatcher::RunnerMixin
+  extend ActiveSupport::Concern
+
+  def log_handle
+    self.class.log_handle
+  end
+
+  def reset_event_monitor_handle
+    @event_monitor_handle = nil
+  end
+
+  def stop_event_monitor
+    @event_monitor_handle.try(:stop)
+  ensure
+    reset_event_monitor_handle
+  end
+
+  def monitor_events
+    event_monitor_handle.start
+    event_monitor_handle.each_batch do |events|
+      event_monitor_running
+      new_events = events.select { |e| whitelist?(e) }
```

Comment:
@moolitayer @jshaughn for a decent description of blacklisting see https://github.com/ManageIQ/manageiq-providers-amazon/issues/140 
So we are about to move the default blacklist to settings and the user added blacklist can be stored in the DB.
If you can do without a whitelist, this would be great. But maybe this can be done after refactoring? Since you will be sharing the code in a mixin...